engine/engine/src/engine.cpp

```diff
@@ -349,6 +349,12 @@ namespace dmEngine
             dmResource::DeleteFactory(engine->m_Factory);
         }
 
+        // Stop processing graphics requests before deleting the graphics context
+        if (engine->m_JobThreadContext)
+        {
+            dmJobThread::Destroy(engine->m_JobThreadContext);
+        }
+
```

Comment:
After fixing the unit test, I saw that we didn't actually destroy the job thread in the engine.